python/desc/imsim/skyModel.py

```diff
@@ -14,21 +14,23 @@
 import galsim
 from lsst.sims.GalSimInterface.galSimNoiseAndBackground import NoiseAndBackgroundBase
 
+from lsst.sims.photUtils import Sed
+
 from .imSim import get_config
 
 __all__ = ['skyCountsPerSec', 'ESOSkyModel', 'get_skyModel_params']
 
 
-# Code snippet from D. Kirkby.  Note the use of astropy units.
-def skyCountsPerSec(surface_brightness=21, filter_band='r',
-                    effective_area=32.4*u.m**2, pixel_size=0.2*u.arcsec):
-    pars = get_skyModel_params()
-    # Lookup the zero point corresponding to 24 mag/arcsec**2
-    s0 = pars[filter_band] * u.electron / u.s / u.m ** 2
-
-    # Calculate the rate in detected electrons / second
-    dB = (surface_brightness - pars['B0']) * u.mag(1 / u.arcsec ** 2)
-    return s0 * dB.to(1 / u.arcsec ** 2) * pixel_size ** 2 * effective_area
+def skyCountsPerSec(skySpec, bandpass, photParams):
+	# Calculate the rate in detected electrons / second.
+  	
+	skySed = Sed(wavelen = skySpec.wave, flambda = skySpec.spec[0,:])
```

Comment:
The `skySpec` object from line 101 below is a tuple.  See the current master of the `sims_skybrightness` package [here](https://github.com/lsst/sims_skybrightness/blob/master/python/lsst/sims/skybrightness/skyModel.py#L634).
Given the way you are using `skySpec` in this function call, you should just pass `skyModel` directly as the first argument of your function call.